### PR TITLE
osx breakpad: getting the gears together for a more advanced local post crash experience.

### PR DESCRIPTION
**Pull request by monkeyiq**
This is for Issue https://github.com/fontforge/fontforge/issues/2120

It has evolved to the point where it can now be merged into mainline.

### CODE REVIEW

File: osx/FontForge.app/Contents/Resources/breakpad/Makefile.am [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+if PLATFORM_OSX
+   ffbpdir = $(pkgdatadir)/osx/FontForge.app/Contents/Resources/breakpad
+   ffbp_DATA = $(FFBP_FILES)
+
+   ffappdir = $(pkgdatadir)/osx/FontForge.app/Contents/Resources/breakpad
+   ffapp_SCRIPTS = $(FFAPP_FILES)
+endif
+
+-include $(top_srcdir)/git.mk
```
Comment (davelab6) [inline osx/FontForge.app/Contents/Resources/breakpad/Makefile.am]: line 43 has a `-` at the start?

File: osx/FontForge.app/Contents/Resources/breakpad/webinterface/Makefile.am [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+WEBINTERFACE_FILES = index.html test.png
+EXTRA_DIST = $(WEBINTERFACE_FILES)
+
+if PLATFORM_OSX
+   webinterfacedir = $(pkgdatadir)/osx/FontForge.app/Contents/Resources/breakpad/webinterface
+   webinterface_DATA = $(WEBINTERFACE_FILES)
+endif
+
+-include $(top_srcdir)/git.mk
```
Comment (davelab6) [inline osx/FontForge.app/Contents/Resources/breakpad/webinterface/Makefile.am]: line 37 has a `-` at the start?

**Comment by davelab6** [comment]
> osx/FontForge.app/Contents/Resources/breakpad/webinterface/test.png

Please replace that with https://github.com/fontforge/fontforge/blob/master/desktop/icons/scalable/apps/fontforge.svg

File: osx/FontForge.app/Contents/Resources/breakpad/Makefile.am (hunk 1 above)
Comment (monkeyiq) [inline osx/FontForge.app/Contents/Resources/breakpad/Makefile.am]: This is a 'try to include but don't complain' version of include;
http://www.jfranken.de/homepages/johannes/vortraege/make_inhalt.en.html#ToC7

It's used throughout, for example the Makefile.am in the root of fontforge:
https://github.com/fontforge/fontforge/blame/master/Makefile.am#L147

I'm not sure if git.mk is possibly unavailable in some cases. git.mk is not in extra_dist for the root directory so it is probably not in make dist; Maybe it makes more sense to have it in make dist and assume we can always find it. Unless the content in it clashes when you don't have a .git directory.

**Comment by monkeyiq** [comment]
cool, will do, and I'll make .svg files and mimetype allowed by the miniserver too :)

**Comment by monkeyiq** [comment]
next up is to mv some thing around so bp on other platforms can share them.
targets:
osx/FontForge.app/Contents/Resources/breakpad/webinterface
osx/FontForge.app/Contents/MacOS/fontforge-crash-reporter.py

and osx/FontForge.app/Contents/Resources/breakpad/minidump_stackwalk will have to be replicated for other platforms. 

Then probably add the option for bp on Linux build.

**Comment by monkeyiq** [comment]
This gives a nice base for a user interface for the breakpad feature on osx.
See https://github.com/fontforge/fontforge/issues/2120

**Pull request merged**